src/Http/BaseApplication.php

```diff
@@ -52,6 +68,63 @@ public function __construct($configDir)
      */
     abstract public function middleware($middleware);
 
+    /**
+     * {@inheritDoc}
+     */
+    public function pluginMiddleware($middleware)
+    {
+        foreach ($this->plugins->with('middleware') as $plugin) {
+            $middleware = $plugin->middleware($middleware);
```

Comment:
This assumes that the `middleware()` method is implemented by the plugin app class, but the interface does not define it. This applies to other hooks as well.
Either we should add hook methods into the `PluginInterface` or check for methods existence inside `with()` method.

Reply:
middleware missing from the interface is a mistake. I'll fix that up.